### DISCUSSION TITLE
dGPU disconnected!

### DISCUSSION THREAD

**Discussion by andyoug** [BartoszCichecki/LenovoLegionToolkit#1150] [Ideas]
Hello,

I've been struggling with a problem on my legion slim 7 gen 8: the fact that it's very difficult to fully disconnect the dGPU for when I'm on the go and need the battery life. A temporary fix for me is to put the laptop to sleep for 1-2 minutes for this to happen, but then, if I restart it, hibernate or plug it in, it automatically reconnects the dGPU.

I tried Toolkit and I'm loving the interface and the multitude of functions, but at first, I only downloaded it because of the disconnect dGPU feature. Unfortunately I found that with it only had 2 options, which don't really help me: "Kill apps" and "Restart dGPU". These don't fully disconnect the gpu, at least in my case. The reason why I'm writing this post and I'm hopeful for a fix is that - when I select the "Restart dGPU" option, the "**dGPU Disconnected**" text appears on screen, that's where I'd need it to stay! So that must mean that Toolkit knows how to disconnect it for good, right?

My idea is to add the option of "Disconnect dGPU" along side "Kill apps" and "Restart dGPU", if it's possible.

Thanks,
Andy

**Comment by BartoszCichecki** [comment]
> but at first, I only downloaded it because of the disconnect dGPU feature

There is no function like this. What you are probably talking about "Deactive GPU" which is not the same.

Getting dGPU to disconnect, ergo making Hybrid-iGPU and Hybrid-Auto to work properly isn't magic, but it's rather quite simple: if your dGPU is used, it the likely hood of it getting ejected is close to 0. That includes apps using the dGPU, but also other apps like Afterburner, HWiNFO64 etc that keep polling dGPU.

Deactivate GPU option is only there to help you get to the point where dGPU is no longer used, as described in the readme. It has nothing to do with Hybrid Modes. The easiest way to tell is check dGPU state in LLT and see it the status is "Powered off". If it doesn't go into powered off, it is used by something.

What you are asking for isn't possible.

**Reply by andyoug** [reply]
Thank you for the response. Yes, I meant "Deactivate".
